src/GafferSceneUI/LightTool.cpp

```diff
@@ -0,0 +1,1266 @@
+const InternedString g_insetPenumbraType( "inset" );
+const InternedString g_outsetPenumbraType( "outset" );
+const InternedString g_absolutePenumbraType( "absolute" );
+const float g_penumbraHandleZAxisRotation = -20.f;
+
+// Return the plug that holds the value we need to edit, and make sure it's enabled.
+
+/// \todo This currently does nothing to enable a row if is disabled. Is that worth doing?
+
+Plug *activeValuePlug( Plug *sourcePlug )
+{
+	if( auto nameValuePlug = runTimeCast<NameValuePlug>( sourcePlug ) )
+	{
+		nameValuePlug->enabledPlug()->setValue( true );
+		return nameValuePlug->valuePlug();
+	}
+	if( auto tweakPlug = runTimeCast<TweakPlug>( sourcePlug ) )
+	{
+		tweakPlug->enabledPlug()->setValue( true );
+		return tweakPlug->valuePlug();
+	}
+	return sourcePlug;
+}
+
+const char *translucentConstantFragSource()
+{
+	return
+		"#if __VERSION__ <= 120\n"
+		"#define in varying\n"
+		"#endif\n"
+		""
+		"in vec3 fragmentCs;"
+		""
+		"void main()"
+		"{"
+		"	gl_FragColor = vec4( fragmentCs, 0.375 );"
+		"}"
+	;
+}
+
+IECoreGL::MeshPrimitivePtr cube()
+{
+	static IECoreGL::MeshPrimitivePtr result;
+	if( result )
+	{
+		return result;
+	}
+
+	IECoreScene::MeshPrimitivePtr mesh = IECoreScene::MeshPrimitive::createBox(
+		Box3f( V3f( -1 ), V3f( 1 ) )
+	);
+
+	ToGLMeshConverterPtr converter = new ToGLMeshConverter( mesh );
+	result = runTimeCast<IECoreGL::MeshPrimitive>( converter->convert() );
+
+	return result;
+}
+
+IECoreScene::MeshPrimitivePtr solidAngle( float radius, float startFraction, float stopFraction, Color3f color )
+{
+	IntVectorDataPtr vertsPerPolyData = new IntVectorData;
+	IntVectorDataPtr vertIdsData = new IntVectorData;
+	V3fVectorDataPtr pData = new V3fVectorData;
+
+	std::vector<int> &vertsPerPoly = vertsPerPolyData->writable();
+	std::vector<int> &vertIds = vertIdsData->writable();
+	std::vector<V3f> &p = pData->writable();
+
+	const int numCircleDivisions = 100;
+	const int numSegments = std::max( 1, (int)ceil( abs( stopFraction - startFraction ) * numCircleDivisions ) );
+
+	p.push_back( V3f( 0, 0, radius ) );
+
+	for( int i = 0; i < numSegments + 1; ++i )
+	{
+		const float a = ( startFraction + ( stopFraction - startFraction ) * (float)i / (float)numSegments ) * 2.f * M_PI;
+		p.push_back( V3f( -sin( a ), 0, -cos( a ) ) * radius + V3f( 0, 0, radius ) );
+	}
+
+	for( int i = 0; i < numSegments; ++i )
+	{
+		vertIds.push_back( i + 1 );
+		vertIds.push_back( i + 2 );
+		vertIds.push_back( 0 );
+		vertsPerPoly.push_back( 3 );
+	}
+
+	IECoreScene::MeshPrimitivePtr solidArc = new IECoreScene::MeshPrimitive( vertsPerPolyData, vertIdsData, "linear", pData );
+	solidArc->variables["N"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Constant, new V3fData( V3f( 0, 1, 0 ) ) );
+	solidArc->variables["Cs"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Constant, new Color3fData( color ) );
+
+	return solidArc;
+}
+
+class LightToolHandle : public Handle
```

Comment:
It would be great to document the basic design intent of LightToolHandle here, with some discussion of how and why responsibilities are divvied up between LightToolHandle and LightTool. It's probably my failing, but I haven't quite got the "big picture" yet.

One thing I'm wondering about is why LightToolThingy _is a_ Handle, rather than why LightToolThingy _has a_ Handle. The original Handle design assumed that the handle provided drag information for a separate client, with the intention that handles could be reused by different clients.

In case it's not clear, I'm not advocating for a redesign at this point, just trying to understand the current design and the reasons behind it.